### PR TITLE
Unexpected ActiveRecord::StatementInvalid

### PR DESCRIPTION
**Pull request by HugsDaniel**
#### What ? Why ?

Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/2537

I replaced the `RecordNotUnique` by `StatementInvalid` with `SQLite3::ConstraintException` to fit the expected `record not unique` error.

### CODE REVIEW

File: core/spec/models/spree/product_spec.rb [hunk 1]
```diff
@@ -222,11 +222,10 @@ def duplicate_extra(old_product)
         end
 
         it "cannot create another product with the same permalink" do
-          pending '[Spree build] Failing spec'
           @product2 = create(:product, :name => 'foo')
           lambda do
             @product2.update_attributes(:permalink => @product1.permalink)
-          end.should raise_error(ActiveRecord::RecordNotUnique)
+          end.should raise_error(ActiveRecord::StatementInvalid, /SQLite3::ConstraintException/)
```
Comment (mkllnk) [inline core/spec/models/spree/product_spec.rb:228]: Is there a more general error we can test here? We actually don't want to use sqlite.

**Review by mkllnk — COMMENTED** [review]
Pau and I discussed at some point that we would like to test with Postgres instead of Sqlite, because that's the database we use in OFN. Can you try to run it with Postgres and see if that solves this spec issue? Do you have access to the Semaphore project to change the configuration?

File: core/spec/models/spree/product_spec.rb (hunk 1 above)
Comment (HugsDaniel) [reply core/spec/models/spree/product_spec.rb:228]: I was able to test with Postgres and it doesn't fail ! Should we keep `AR::RecordNotUnique` then ?

Comment (mkllnk) [reply core/spec/models/spree/product_spec.rb:228]: Yes, the fewer patches we need, the better. :-) Thank you for testing.

**Comment by HugsDaniel** [comment]
This can be closed since #17 solve the issue.

**Pull request closed without merging**